upload_multiple_files_to_single_record.py

```diff
@@ -0,0 +1,224 @@
+    url = "{}?{}".format(url, params)
+
+    headers = {'Authorization': 'Token ' + API_KEY,
+               'Content-Type': 'application/xml'}
+
+    r = requests.post(url, headers=headers, data=string_xml)
+
+    print("Metadata upload status: %s" % (r.text,))
+
+
+def upload_and_save_xml(collection):
+    """
+    Upload the MARCXML with the record API and save a local copy of the file.
+    """
+    # Convert to string, replace
+    string_xml = tostring(collection, pretty_print=True, encoding="unicode")
+
+    # Replace capital TAG to lowercase tag in data fields
+    string_xml = string_xml.replace('datafield TAG="', 'datafield tag="')
+
+    string_xml_name = data_folder.parent / 'export_{}.xml'.format(data_folder.name)
+
+    # Save xml file
+    with open(string_xml_name, 'w') as xmlfile:
+        xmlfile.write(string_xml)
+
+    # Upload xml file
+    try:
+        upload_metadata(string_xml)
+    except UnicodeEncodeError as e:
+        print("Cannot import XML file: %s. Error message: \n %s" % (string_xml_name, e))
+
+
+if __name__ == '__main__':
+    # Get the path to the files.
+    folder_path = input('Insert folder path for the files to upload:\n')
+    folder_path = folder_path.replace('\\', '').strip()
+    data_folder = Path(folder_path)
+    if not data_folder.is_dir():
+        print('The provided path is not a folder. Try again!')
+        exit()
+
+    # Create xml record object
+    new_record = E.record()
+
+    # Append metadata to record
+    for key, val in APPEND_METADATA.items():
+        if len(key) == 6:
+            new_record.append(
+                create_datafield(key[0:5],
+                                 [(key[5], val)]))
+
+    # Loop through all files in path, except DS_Store (MacOS specific files).
+    for i, file_path in enumerate(sorted(data_folder.rglob('*.[!DS_Store]*'))):
+
+        # Ignore directories
+        if file_path.is_dir():
+            continue
+
+        # Step 1: Get AWS presigned object
+        # Try to request the preseign object five times before going to the next file.
+        presigned_response = None
+        connections = 0
+        while presigned_response is None and connections < 5:
+            try:
+                presigned_response = request_presigned_object()
+            except ConnectionResetError:
+                connections += 1
+        if presigned_response is None:
+            continue
+
+        # Step 2: Upload file
+        try:
+            upload_response = upload_single_file(str(file_path), presigned_response)
+        except:
+            print("File failed to upload: %s" % (file_path,))
+            continue
+
+        # Step 3: Get local checksum
+        local_checksum = get_md5_checksum(str(file_path))
+
+        # Step 4: Compare the checksum of local file and uploaded file
+        verified = compare_checksum(local_checksum, upload_response)
+
+        # Step 5: Create FFT datafield and attach to record.
+        if verified:
```

Comment:
Maybe something like this is even clearer:

```suggestion
        # Step 3: Get local checksum
        local_checksum = get_md5_checksum_of_file(str(file_path))

        # Step 4: Get remote checksum
        remote_checksum = get_md5_checksum_from_response(upload_response)

        # Step 5: Compare checksums, create FFT datafield and attach to record.
        if local_checksum == remote_checksum:
```